Calculate percentage difference between two values

christippett opened the pull request:
This is a calculation I use quite often and make use of as a UDF. I thought it'd be useful for others too, hence the PR.

christippett commented:
@googlebot I signed it!

ryanmcdowell commented on udfs/community/percentage_difference.sql:
```diff
@@ -0,0 +1,21 @@
+  CASE WHEN val1 > 0 AND val2 > 0 THEN 
+    ABS(val1 - val2) / ((val1 + val2) / 2)
+  END
+);
```
Add trailing newline

christippett commented:
Thanks for the feedback @ryanmcdowell.

I've added test cases that cover different data type input scenarios, as well as a few others. I've also updated the README.

Note: I've rounded the output with an opinionated precision of `4`, which I think is reasonable for a percentage value - especially as the output is a FLOAT64 value. What do you think?
